doc/ir-semantics.text

```diff
@@ -110,10 +111,18 @@ Update (partial) function, returns an H:
 
 ## Osr transformation of environment
 
-  osr-env P H E (const x = e) = (x ↦ eval P H E e)
-  osr-env P H E (mut x)       = (x ↦ ⊥)
-  osr-env P H E (mut x = y)   = (x ↦ a)
+  materialize P H E (mut x = e) = (x ↦ (a ↦ v))
+    where a - fresh
+      and v := eval P H E e
+  materialize P H E _           = ()
+
+  osr-env P H E M (const x = e) = (x ↦ val v)
+    where v := eval P H E e
+  osr-env P H E M (mut x)       = (x ↦ ⊥)
+  osr-env P H E M (mut x = &y)  = (x ↦ ref a)
     where a = E(y)
+  osr-env P H E M (mut x = e)   = (x ↦ ref e)
```

Comment:
Oh, I think I get it. `e` is evaluated by `materialize`.

But then it should be `(x ↦ ref a)`, right?